RELEASE_PROCEDURE.md

```diff
@@ -25,14 +25,30 @@ DLA-Future follows [Semantic Versioning](https://semver.org).
 1. Add a link to the documentation for the release in `DOCUMENTATION.md` and update the link in `README.md`.
    The documentation will be generated automatically after the `vx.y.z` tag has been created and pushed.
 
+1. Merge all the changes to the `master` branch (also for patch releases).
+
 1. Ensure you have `gh` ([GitHub CLI](https://cli.github.com)) installed. Run `gh auth login` to authenticate
    with your GitHub account, or set `GITHUB_TOKEN` to a token with `public_repo` access.
 
+1. Ensure you are working from the `eth-cscs/DLA-Future` and not your own fork for following steps.
+
+1. For patch releases: switch to `version_X.Y` branch.
+
+1. For patch releases: cherry pick the commits for the patch from `master` to the `version_X.Y` branch.
+   Make sure the cherry-picked commits are listed in the `CHANGELOG.md`, and make sure you cherry-picked
+   the PR(s) updating all the above (version, documentation, etc.).
+
+1. Check the release branch to make sure the content matches the release notes.
+
 1. Create a release on GitHub using the script `scripts/roll_release.sh`. This
    script automatically tags the release with the corresponding release number.
```

Comment:
I think @rasolca's comment about waiting for CI to pass before making the release is still missing from here (https://github.com/eth-cscs/DLA-Future/pull/1250#issuecomment-2554851068).

Reply:
Good catch! It was lost in 862a9a3f86d5fc0a8af00f7ccd00442838629941 when I changed to always push everything to master. I'll re-add it. I also double checked that the CI is automatically triggered for the `version_*` branches, which it is.